web/src/stores/websocket.ts

```
@@ -0,0 +1,228 @@
+import { defineStore } from 'pinia'
+import { ref } from 'vue'
+import type { WsChannel, WsEvent, WsEventHandler } from '@/api/types'
+import { WS_RECONNECT_BASE_DELAY, WS_RECONNECT_MAX_DELAY, WS_MAX_RECONNECT_ATTEMPTS } from '@/utils/constants'
+
+/** Strip all control characters and truncate for safe logging. */
+function sanitizeLogValue(value: unknown, max = 200): string {
+  return String(value).replace(/[\x00-\x1f\x7f]/g, ' ').slice(0, max)
```

Biome (2.4.6)

[error] Unexpected control character in a regular expression.

(lint/suspicious/noControlCharactersInRegex)